core/drand_daemon_helper.go

```diff
@@ -44,9 +44,13 @@ func (dd *DrandDaemon) readBeaconID(metadata *protoCommon.Metadata) (string, err
 	}
 
 	// if we didn't match on a chain hash, and have the empty string, then it's the default beacon
-	if rcvBeaconID == "" {
-		rcvBeaconID = common.DefaultBeaconID
+	rcvBeaconID = common.GetCorrectBeaconID(rcvBeaconID)
+	// make sure the metadata use a correct beacon id
+	if metadata == nil {
+		metadata = &protoCommon.Metadata{}
 	}
+	// we explicitly set the beacon id on the metadata in case it changed because of GetCorrectBeaconID
+	metadata.BeaconID = rcvBeaconID
```

Comment:
Mhhh it's unclear why do set previously the `metadata.BeaconID` before, if the if/else if we anyway overwrite it here. It feels the `rcvBeaconID` is the one important here.
Also can we change the
```
if rcvBeaconID != "" && !common.CompareBeaconIDs(rcvBeaconID, beaconIDByHash) {
```
to using defined functions in the set of function that manage/check beacons ID ? (instead of checking "") - I know it's not in this PR but if we can really remove all references to hardcoded values that would be great.

Reply:
The thing is: if the received beacon ID is empty, it might be the case that it's meant to be empty because only the chainHash is known, and set.

So it really makes sense to check it here, in that function, no?

Reply:
I don't follow. The  simplified flow looks like this:
```
* If chainhash is not empty, then  
    + Check for stuff and overwrite `metadata.BeaconID` (or if error, return) 
* Overwrite `metadata.BeaconID`
```
What I'm saying is we overwrite anyway `metadata.BeaconID` and that looks weird. We should probably have a full `if chianhash.len() == 0 { ... } else { ... } ` or stg no ? Or remove the fact we write to `metadata.BeaconID` inside the `if/else` ?

Reply:
I removed the fact that we write to it in the first part of the `if`, is that not what you meant?